From 560ac05130696de2491881bbc2a5024c94bc3912 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Tue, 22 Jun 2021 16:16:45 +0900
Subject: [PATCH] ksmbd: remove unneeded NULL check in the list iterator
Git-commit: 560ac05130696de2491881bbc2a5024c94bc3912
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Remove unneeded NULL check in the list iterator. And use
list_for_each_entry_safe instead of list_for_each_safe.

Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/mgmt/user_session.c | 15 ++++++---------
 fs/cifsd/smb2pdu.c           |  2 +-
 2 files changed, 7 insertions(+), 10 deletions(-)

diff --git a/fs/cifsd/mgmt/user_session.c b/fs/cifsd/mgmt/user_session.c
index 615b46f0762b..c5ba9694e1f1 100644
--- a/fs/cifsd/mgmt/user_session.c
+++ b/fs/cifsd/mgmt/user_session.c
@@ -30,15 +30,12 @@ struct ksmbd_session_rpc {
 
 static void free_channel_list(struct ksmbd_session *sess)
 {
-	struct channel *chann;
-	struct list_head *tmp, *t;
-
-	list_for_each_safe(tmp, t, &sess->ksmbd_chann_list) {
-		chann = list_entry(tmp, struct channel, chann_list);
-		if (chann) {
-			list_del(&chann->chann_list);
-			kfree(chann);
-		}
+	struct channel *chann, *tmp;
+
+	list_for_each_entry_safe(chann, tmp, &sess->ksmbd_chann_list,
+				 chann_list) {
+		list_del(&chann->chann_list);
+		kfree(chann);
 	}
 }
 
diff --git a/fs/cifsd/smb2pdu.c b/fs/cifsd/smb2pdu.c
index 341d51e711a5..bbb35e68abc4 100644
--- a/fs/cifsd/smb2pdu.c
+++ b/fs/cifsd/smb2pdu.c
@@ -76,7 +76,7 @@ struct channel *lookup_chann_list(struct ksmbd_session *sess, struct ksmbd_conn
 	struct channel *chann;
 
 	list_for_each_entry(chann, &sess->ksmbd_chann_list, chann_list) {
-		if (chann && chann->conn == conn)
+		if (chann->conn == conn)
 			return chann;
 	}
 
-- 
2.33.0

